test/eventlistener_test.go

```diff
@@ -0,0 +1,114 @@
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"testing"
+
+	knativetest "github.com/knative/pkg/test"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestEventListenerCreate(t *testing.T) {
+	c, namespace := setup(t)
+	t.Parallel()
+
+	defer tearDown(t, c, namespace)
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+
+	t.Log("Start EventListener e2e test")
+
+	// Create EventListener
+	el, err := c.TriggersClient.TriggersV1alpha1().EventListeners(namespace).Create(
+		&v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-eventlistener",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				TriggerBindingRefs: []v1alpha1.TriggerBindingRef{
+					{
+						Name: "my-triggerbinding",
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		t.Fatalf("Failed to create EventListener: %s", err)
+	}
+	t.Logf("Created EventListener %s in namespace %s", el.Name, el.Namespace)
+
+	// Verify the EventListener's Deployment is created
+	if err = wait.PollImmediate(interval, timeout, func() (bool, error) {
+		_, err := c.KubeClient.AppsV1().Deployments(namespace).Get(el.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		return true, err
+	}); err != nil {
+		t.Fatalf("Failed to create EventListener Deployment: %s", err)
+	}
+	t.Logf("Found EventListener's Deployment")
+
+	// Verify the EventListener's Service is created
+	if err = wait.PollImmediate(interval, timeout, func() (bool, error) {
+		_, err := c.KubeClient.CoreV1().Services(namespace).Get(el.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		return true, err
+	}); err != nil {
+		t.Fatalf("Failed to create EventListener Service: %s", err)
+	}
+	t.Logf("Found EventListener's Service")
+
+	// Delete EventListener
+	err = c.TriggersClient.TriggersV1alpha1().EventListeners(namespace).Delete(el.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("Failed to delete EventListener: %s", err)
+	}
+	t.Logf("Deleted EventListener")
+
+	// Verify the EventListener's Deployment is deleted
+	if err = wait.PollImmediate(interval, timeout, func() (bool, error) {
+		_, err := c.KubeClient.AppsV1().Deployments(namespace).Get(el.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	}); err != nil {
+		t.Fatalf("Failed to delete EventListener Deployment: %s", err)
+	}
```

Comment:
sometimes it can make the tests a bit cleaner to move some of this polling logic into a lib like https://github.com/tektoncd/pipeline/blob/42bc3fbd5fc3ed35bfad1f371ef585da92b5fdc6/test/wait.go

but probably not a lot cleaner, and also less valuable if we don't think folks would use the functions in other tests to so kind of a toss up! I'll leave it to your preferences :)

Reply:
I created one for now. Although given that we'll probably be [polling on the EventListener's `Status`](https://github.com/tektoncd/triggers/pull/53#discussion_r309023062) in the future, I don't know if these functions will stay around for too long 😄